syntax = "proto3";

package messages;

message Base
{
    uint32 seq_num = 1;
}

message PingReq
{
    Base base = 1;
}

message PingRsp
{
    Base base = 1;
    uint32 req_seq_num = 2;
}

message Text
{
    Base base = 1;
    string msg_txt = 2;
}

message Goodbye
{
    Base base = 1;
    enum Reason 
    {
        UNKNOWN = 0;
        SERVER_SHUTDOWN = 1;
    }
    Reason reason = 2;
    string reason_str = 3;
}